invoices/invoiceregistry.go

```diff
@@ -1043,3 +1141,309 @@ func (i *InvoiceRegistry) HodlUnsubscribeAll(subscriber chan<- interface{}) {
 
 	delete(i.hodlReverseSubscriptions, subscriber)
 }
+
+// updateResult is the result of the invoice update call.
+type updateResult uint8
+
+const (
+	resultReplayToCanceled updateResult = iota
```

Comment:
perhaps add a `resultInvalid` state for 0, to prevent accidental initialization with `resultReplayToCanceled`?

Reply:
added